src/main/java/seedu/duke/command/EditCommand.java

```diff
@@ -135,57 +138,73 @@ public void execute(TransactionList transactions, Ui ui, Storage storage) throws
             int newAmount = amount;
             LocalDate newDate = date;
             String newCategory = category;
-            Transaction entry = transactions.getEntry(index - 1);
 
             boolean check = isIndexValid(transactions, index);
             boolean secondCheck = isTagsNonEmpty(newType, newDescription, newAmount, newCategory, newDate);
-            boolean thirdCheck = isParametersChanged(entry, newType, newDescription, newAmount, newCategory, newDate);
-            assert index > 0;
-
-            if (check && secondCheck && thirdCheck) {
-                newType = updateType(entry, newType);
-
-                if (newType.equals("expense")) {
-                    newDate = updateDate(entry, newDate);
-                    newDescription = updateDescription(entry, newDescription);
-                    newCategory = updateCategory(entry, newCategory);
-                    newAmount = updateAmount(entry, newAmount);
 
-                    String message = transactions.editExpense(newDescription, newAmount, newCategory, newDate, index);
-
-                    long addedMonthExpenseSum = transactions.calculateMonthlyTotalExpense(newDate);
-                    String budgetInfo = Budget.generateBudgetRemainingMessage(addedMonthExpenseSum, true,
-                            DateFormats.retrieveFormattedMonthAndYear(newDate));
+            assert index > 0;
 
-                    Ui.showTransactionAction(INFO_EDIT_EXPENSE.toString(), message, budgetInfo);
-                    editLogger.log(Level.INFO, "The requested transaction has been edited "
-                            + "and the UI should display the confirmation message respectively.");
+            editLogger.log(Level.INFO, "If the index is invalid, the command will not execute and throw "
+                    + "an exception later.");
+            if (check) {
+                Transaction entry = transactions.getEntry(index - 1);
+                boolean thirdCheck = isParametersChanged(entry, newType, newDescription, newAmount,
+                        newCategory, newDate);
+
+                editLogger.log(Level.INFO, "If tags are empty or the tags will result in an unedited "
+                        + "transaction, the command will stop executing and throw an exception later.");
+
+                if (secondCheck && thirdCheck) {
```

Comment:
Might want to name this variables with semantic meanings as it isn't quite clear what are these checks